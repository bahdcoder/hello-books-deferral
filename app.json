{
  "name": "Hello Books",
  "description": "A system that allows efficient, cost-effective management of Books for Libraries."
}
