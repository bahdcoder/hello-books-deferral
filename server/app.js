import express from 'express';
import consola from 'consola';
import bodyParser from 'body-parser';
import swagger from 'swagger-ui-express';
import Knex from 'knex';
import { createServer } from 'http';
import { Model } from 'objection';
import swaggerDocument from '@/swagger.json';
import Routes from '@routes/v1';
import config from '@config';

import 'express-jsend';

export const app = express();

export const databaseConnection = Knex(config.database[config.server.env]);

Model.knex(databaseConnection);

app.use(bodyParser.json());
app.use(bodyParser.urlencoded({ extended: true }));

Routes(app);

app.get('/', (req, res) => {
    res.status(200).jsend({ message: 'Hello Books Deferral' });
});
/**
 *  setup Swagger
 */
app.use(
    '/api-docs',
    swagger.serve,
    swagger.setup(swaggerDocument, { explorer: true })
);

export const server = createServer(app);

export default { app, databaseConnection };
