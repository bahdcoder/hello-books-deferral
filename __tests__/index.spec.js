import { createServer } from 'http';
import supertest from 'supertest';
import { server as app } from '@server/app';

const server = () => supertest(app);

describe('APP HOST /', () => {
    test('should return Hello Books Deferral', async () => {
        const { status, body } = await server().get('/');

        expect(status).toBe(200);
        /* expect(body).toMatchSnapshot(); */
    });
});
