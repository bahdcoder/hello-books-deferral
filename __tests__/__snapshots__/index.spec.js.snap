// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APP HOST / should return Hello Books Deferral 1`] = `
Object {
  "data": Object {
    "message": "Hello Books Deferral",
  },
  "status": "success",
}
`;
